tests/framework/storage.go

```diff
@@ -0,0 +1,294 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/kubernetes"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"kubevirt.io/client-go/kubecli"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+)
+
+const (
+	pollInterval = 3 * time.Second
+	waitTime     = 600 * time.Second
+)
+
+func CreateDataVolumeFromDefinition(clientSet kubecli.KubevirtClient, namespace string, def *cdiv1.DataVolume) (*cdiv1.DataVolume, error) {
+	var dataVolume *cdiv1.DataVolume
+	// TODO: use cdiclient.Interface
+	err := wait.PollImmediate(pollInterval, waitTime, func() (bool, error) {
+		var err error
+		dataVolume, err = clientSet.CdiClient().CdiV1beta1().DataVolumes(namespace).Create(context.TODO(), def, metav1.CreateOptions{})
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		return false, err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return dataVolume, nil
+}
+
+// FindPVC Finds the passed in PVC
```

Comment:
Nitpick, but not too sure if `passed in` makes sense in this context. This is repeated several times along the PR.